agenda/data/src/main/java/com/gkp/agenda/data/workers/CreatedItemsWorker.kt

```diff
@@ -0,0 +1,54 @@
+package com.gkp.agenda.data.workers
+
+import android.content.Context
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import com.gkp.agenda.data.getType
+import com.gkp.agenda.data.toReminderBody
+import com.gkp.agenda.data.toTaskBody
+import com.gkp.agenda.domain.datasource.LocalAgendaDataSource
+import com.gkp.agenda.domain.model.AgendaItem
+import com.gkp.agenda.domain.model.AgendaItemType
+import com.gkp.core.database.dao.CreatedAgendaItemsDao
+import com.gkp.core.network.TaskyRetrofitApi
+
+class CreatedItemsWorker(
+    private val taskyRetrofitApi: TaskyRetrofitApi,
+    private val localDataSource: LocalAgendaDataSource,
+    private val createdAgendaItemsDao: CreatedAgendaItemsDao,
+    context: Context,
+    params: WorkerParameters,
+) : CoroutineWorker(context, params) {
+
+    override suspend fun doWork(): Result {
+        val agendaItemId = inputData.getString(AGENDA_ITEM_ID) ?: return Result.failure()
+        val agendaType = inputData.getString(AGENDA_ITEM_TYPE) ?: return Result.failure()
+        return try {
+            val agendaItem = localDataSource.getAgendaItemById(agendaItemId)
+            performPushCreatedRequest(agendaItem)
+            createdAgendaItemsDao.deleteById(agendaItemId)
+
+            Result.success()
+        } catch (e: Exception) {
+            Result.failure()
+        }
```

Comment:
Be careful with catching general exceptions inside suspend functions. You can break coroutine cancellation that way. If a coroutine is cancelled, the currently suspending function (here, it would be `FUNCTION_NAME`) will throw a CancellationException. If your catch block eats that up, this exception won't be properly propagated up the coroutine scopes which can cause some coroutines to run longer than they should.
You can fix this by adding this:

```
if(e is CancellationException) throw e
```
or alternatively by only catching specific exceptions and not CancellationExceptions.

In this video Philipp shows the issue with that a bit more in detail (starting at 17:45):
https://www.youtube.com/watch?v=VWlwkqmTLHc